test/export/test_export.py

```diff
@@ -7242,6 +7240,41 @@ def forward(self, spec):
         spec = torch.complex(real_part, imaginary_part)
         export(model, (spec,))
 
+    def test_export_linear_preserve_dynamic_shape(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.lin = torch.nn.Linear(4, 4)
+
+            def forward(self, x):
+                return self.lin(x)
+
+        mod = M()
+        ep = export(
+            mod,
+            (torch.randn(8, 4),),
+            dynamic_shapes={
+                "x": {
+                    0: Dim("x"),
+                }
+            },
+        )
+
+        if IS_FBCODE:
+            ep = ep.run_decompositions(
+                {}, _preserve_ops=(torch.ops.aten.linear.default,)
```

Comment:
what's the plan for unifying the fbcode path?

Reply:
https://www.internalfb.com/diff/D62906375 I have a PR here to remove this. I will probably land it this week.